pylint/checkers/typecheck.py

```diff
@@ -1202,24 +1178,24 @@ def _get_nomember_msgid_hint(
         node: nodes.Attribute | nodes.AssignAttr | nodes.DelAttr,
         owner: SuccessfulInferenceResult,
     ) -> tuple[Literal["c-extension-no-member", "no-member"], str]:
-        suggestions_are_possible = self._suggestion_mode and isinstance(
-            owner, nodes.Module
+        if isinstance(owner, nodes.Module) and _is_c_extension(owner):
```

Comment:
_is_c_extension already checks isinstance(..., Module).

Reply:
Nice catch ! Done in [8e103a6](https://github.com/pylint-dev/pylint/pull/9962/commits/8e103a6b98a6046bb6984729a3af5482605aadf5)